### PR TITLE
connection between cli and server

### PR DESCRIPTION
**Pull request by armintalaie**
List of commands from the CLI

particularly,
- send {} will send anything after the command
- help or any wrong command will display list of commands
- quit will quit program

program makes connection to the server and actively listens but only sends if followed by the send command for now

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -1,62 +1,128 @@
...
 async fn main() {
-    let connect_addr = "ws://localhost:3000";
-    let url = url::Url::parse(&connect_addr).unwrap();
+    let (stdin_tx, stdin_rx) = mpsc::unbounded::<Message>();
+    let (write, read) = server_connection().await;
+
+    let stdin_to_ws = tokio::spawn(write_text_to_server(write, stdin_rx));
+    let listen_stream = tokio::spawn(listen_to_server(read));
+
+    let cli_sender = stdin_tx.clone();
```
Comment (bonvee-99) [inline cli/src/main.rs]: we dont need to clone this unless we want to have another producer. In this case there's a single producer

File: cli/src/main.rs [hunk 2]
```diff
@@ -1,62 +1,128 @@
...
+}
+
+async fn listen_to_server(read: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>) {
+    read.for_each(|message| async {
+        let data = message.unwrap().into_data();
+        println!("\x1b[1m\x1b[95m>>\x1b[0m");
+        println!("\x1b[1m\x1b[36m{:?}\x1b[0m", String::from_utf8(data).unwrap());
+        println!("\x1b[1m\x1b[95m<<\x1b[0m");
+        // let data2 = data.clone();
+        // println!("got message: {:?}", String::from_utf8(data2).unwrap());
```
Comment (bonvee-99) [inline cli/src/main.rs]: can we remove comments

File: cli/src/main.rs [hunk 3]
```diff
@@ -1,62 +1,128 @@
...
-
-    let stdin_to_ws = stdin_rx.map(Ok).forward(write);
-    let ws_to_stdout = {
-        read.for_each(|message| async {
-            let data = message.unwrap().into_data();
-            // let data2 = data.clone();
-            // println!("got message: {:?}", String::from_utf8(data2).unwrap());
-            tokio::io::stdout().write_all(&data).await.unwrap();
-            tokio::io::stdout().flush().await.unwrap();
-            // std::io::stdout().write_all(&data2).unwrap();
```
Comment (bonvee-99) [inline cli/src/main.rs:41]: remove comments?

**Review by bonvee-99 — COMMENTED** [review]
nice

**Comment by armintalaie** [comment]
@bonvee-99 plz approve and merge if everything else looks good. thx.

**Pull request merged**